CHANGELOG.md

```diff
@@ -1,7 +1,49 @@
 # Change Log
 All notable changes to the "vscode-azurearmtools" extension will be documented in this file.
 
-## Version 0.8.0 (2019-10-01)
+## Coming Soon
+- Support for variable iteration ("COPY blocks")
+
+## Version 0.8.0 (2019-10-28)
+### Added
+- Greatly improved schema errors and completion
+  - Our new language server now has a better understanding of Azure Resource Manager templates and can therefore provide a better error and completion experience beyond that provided using standard JSON validation
+  - It now narrows down its schema validation based on the resource type and apiVersion properties of each resource.
+  - No longer suggests changing the resource type if the apiVersion is not supported, or if a property is missing or invalid
+  - Errors provide better guidance for what actually needs to be fixed in each resource to match the schema
+  - Greatly improved completion speed
+  - Greatly reduced memory usage
+  - Note: The Azure schemas should continue to improve (this is a separate issue from this extension)
+- Full support for user-defined template functions [#122](https://github.com/microsoft/vscode-azurearmtools/issues/122). See Azure [documentation](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-authoring-templates#functions).
+  - Find References (namespaces, user functions, user function parameters)
+  - Go to Definition
+  - Expression completion
+  - Parameter count validation
+  - Rename user function parameters
+  - Hover information
+  - Signature help
+- Added basic snippet support. This makes Sam Cogen's [Azure Resource Manager Snippets](https://marketplace.visualstudio.com/items?itemName=samcogan.arm-snippets) extension no longer necessary when using this extension.  If you have snippet suggestions, you can add them to our [repo](https://github.com/microsoft/vscode-azurearmtools/issues).
+- Hover information for JSON properties (outside of expressions)
+- Fewer false positives
+  - We now allow string values to be automatically coerced to non-string types as appropriate (e.g., "false" can be used in place of false), as the ARM backend allows
+  - We now allow non-string values to be automatically coerced to string as appropriate (e.g. false can be used in place of "false")
+
```

Comment:
- Added broader support for ARM deployment scopes (resource group, subscription, management group, tenant).  That was new for some of our features at least.

Reply:
Unfortunately the change on Lidong's side to support these had to be reverted, so we get validation errors if these are used.  Will add in 0.8.1.